frontend/packages/dev-console/console-extensions.json

```diff
@@ -685,6 +685,20 @@
       "insertAfter": "console.preferredCreateEditMethod"
     }
   },
+  {
+    "type": "console.user-preference/item",
+    "properties": {
+      "id": "console.preferredResource",
+      "label": "%devconsole~Resource type%",
+      "groupId": "applications",
+      "description": "%devconsole~If resource type is not selected, the console default to the latest.%",
+      "field": {
+        "type": "custom",
+        "component": { "$codeRef": "userPreferences.ResourceType" }
```

Comment:
Is a custom component necessary?
Why not do this declaratively using flags if knative is installed or not?

Reply:
@christianvogt I used the custom component so that I can show the Serverless Deployment option based on if knative is installed.

We can use the type `dropdown` field. For that, we have to define this extension with the same id in `dev-console` and in `knative-plugin`  with `Serverless deployment` option and with the required flag KNATIVE_SERVING. And show the one Resource Type dropdown at a moment we need to implement priority for `console.user-preference/item`.

Is this what you mean?

Reply:
Essentially, yes.

You can keep both definitions in dev-console since dev-console has knowledge of knative for import already. You'd create 2 extension points with similar content but one would also include serverless option. One would disallow `KNATIVE_SERVING` flag while the other would require `KNATIVE_SERVING` flag to keep them mutually exclusive.